lib/injectable/call_facade.rb

```diff
@@ -0,0 +1,26 @@
+module Injectable
+  module CallFacade
+    def self.[](klass)
+      m = Module.new do
+        # Entry point of the service.
+        # Arguments for this method should be declared explicitly with '.argument'
+        # and declare this method without arguments
+        define_method :call do |args = {}|
+          if self.instance_of?(klass)
+            check_call_definition!
+            check_missing_arguments!(self.class.required_call_arguments, args)
+            variables_for!(self.class.call_arguments, args)
+          end
+
+          super()
+        end
+
+        def self.facade?
```

Comment:
Here we can use different ways to recognize a module as a facade. It's anyways important, otherwise `check_call_definition!` will give false positives.